qutebrowser/config/configfiles.py

```diff
@@ -222,20 +223,43 @@ def read_config_py(filename=None):
         raise configexc.ConfigFileErrors(basename, [desc])
     except SyntaxError as e:
         desc = configexc.ConfigErrorDesc("Syntax Error", e,
-                                         traceback=traceback.format_exc())
+                                        traceback=traceback.format_exc())
         raise configexc.ConfigFileErrors(basename, [desc])
 
     try:
-        exec(code, module.__dict__)
+        # Save and restore sys variables
+        with saved_sys_properties():
+            # Add config directory to python path, so config.py can import
+            # other files in logical places
+            config_dir = os.path.dirname(filename)
+            if config_dir not in sys.path:
+                sys.path = [config_dir] + sys.path
```

Comment:
`sys.path.insert(0, config_dir)` maybe?

Reply:
I can either make a copy (via `sys.path.copy()` or similar) when save/restoring, or I can avoid touching this variable inside the `when`. I think that the copy is better, since then whoever is using the contextmanager dosen't need to worry about modifying the original list, so I'll stick to that.

Reply:
I don't understand your comment - wouldn't `sys.path = [config_dir] + sys.path` and `sys.path.insert(0, config_dir)` do the same here?